echo_server.py

```diff
@@ -3,6 +3,58 @@
 import socket
 
 
+def response_ok():
+    """Return byte string 200 ok response."""
+    return u"HTTP/1.1 200 OK\r\nContent-Type: text/plain\r\nContent-length: 18\r\n\r\neverything is okay".encode('utf-8')
+
+
+def response_error(error_code, reason):
+    """Return byte string error code."""
+    return u"HTTP/1.1 {} {}\r\nContent-Type: text/plain\r\nContent-length: {}\r\n\r\n{}".format(error_code, reason, len(reason), reason).encode('utf-8')
+
+
+# def parse_request(request):
+#     """Parse HTTP request and return URI requested.
+#     If not GET requests, raises error.
+#     If not HTTP/1.1 request, raises error."""
+#     first_line = request.split("\n")[0].split()
+#     if first_line[2] == 'HTTP/1.1':
+#         if first_line[0] == 'GET':
+#             return first_line[1]
+#         else:
+#             return response_error(405, 'Method Not Allowed')
+#     else:
+#         return response_error(505, 'HTTP Version Not Supported')
+
+
+class Error405(BaseException):
+    def __init__(self):
+        self.out = response_error(405, 'Method Not Allowed')
+
+    def __str__(self):
+        return self.out
+
+
+class Error505(BaseException):
+    def __init__(self):
+        self.out = response_error(505, 'HTTP Version Not Supported')
+
+    def __str__(self):
+        return self.out
+
+
+def parse_request2(request):
+    """Parse HTTP request and returns first line as a list."""
+    return request.split("\n")[0].split()
+
+
+def check_request(first_line):
+    if first_line[2] != 'HTTP/1.1':
+        raise Error505
+    if first_line[0] != 'GET':
+        raise Error405
+    return response_ok()
```

Comment:
The combination of `parse_request2` and `check_request` make up what `parse_request` is supposed to do on its own, with one exception.  It should return the URI, not a response.